src/chapters/classes/encapsulation.rst

```diff
@@ -0,0 +1,289 @@
+Encapsulation
+=============
+
+Our discussion of classes and objects is integral to us using **object-oriented programming**.
+Object-oriented programming stands on four pillars: abstraction, encapsulation, inheritance, and polymorphism.
+
+.. index:: ! encapsulation, ! instance variable
+
+Encapsulation
+-------------
+
+**Encapsulation** is the bundling of related data and behaviors that operate on that data, 
+usually with restricted access to internal, non-public data and behaviors.
+In object-oriented programming, classes and objects allow us to encapsulate, or isolate, 
+data and behavior to only the parts of our program to which they are relevant.
+Restricting access allows us to expose only that data and behavior that we want others to be able to use.
+
+Let's take a look at this by developing a new class called ``Student``.
+
+``Student`` Class
+-----------------
+
+Fields
+^^^^^^
+
+We previously defined a **field** as a variable, or piece of data, that
+belongs to a class. For our ``Student`` class, let’s think about the
+data that is typically associated with a student (in the sense of a high
+school or college student). There are a lot of possibilities, but here
+are the most important:
+
+1. Name
+2. Student ID
+3. Number of credits
+4. GPA
+
+In order to declare these fields within our class, we’ll need to
+determine the best data type for each. A field may be of any primitive
+or object type. In this case, the following types will work best:
+
+1. Name: ``string``
+2. Student ID: ``int``
+3. Number of credits: ``int``
+4. GPA: ``double``
+
+Let’s put these inside of a class. While they may be declared anywhere
+within a class, fields should always be declared at the top of the
+class. When we’re ready to add methods, we’ll add them below the fields.
+
+.. sourcecode:: c#
+   :linenos:
+
+   public class Student 
+   {
+      string name;
+      int studentId;
+      int numberOfCredits;
+      double gpa;
+   }
+
+Like variables within a method, fields may be initialized when they are
+declared. For example, we could provide default values for
+``numberOfCredits`` and ``gpa`` (default values for ``name`` and
+``studentId`` don’t make sense since they should be different for each
+student).
+
+.. sourcecode:: c#
+
+   int numberOfCredits = 0;
+   double gpa = 0.0;
+
+.. index:: ! instance variable
+
+Fields are also referred to as **instance variables**, since they belong
+to an instance of a class.
+
+.. index:: ! accessor, ! getter, ! setter, ! auto-implemented property, ! backing field
+
+Getters and Setters
+^^^^^^^^^^^^^^^^^^^
+
+As declared, our four fields are private, which means that they are inaccessible 
+to code outside of the ``Student`` class. As a
+rule-of-thumb, *fields should always be private unless you have a very,
+very, very good reason to not make them so.* As we mention on the previous page, it is 
```

Comment:
I changed some of this up more along the lines of this comment than the next one. I like the idea to put this above the section header to help with the transition rather than adding it to the end of the section because a) we need an explanation for why we use getters/settters and thats harder without already discussing why fields should be private. and b) this section is long and the end is closer to the auto-implemented property which draws upon a more implicit private field which could be confusing imo